React/Base/RCTMultipartStreamReader.m

```diff
@@ -56,14 +65,34 @@ - (void)emitChunk:(NSData *)data callback:(RCTMultipartCallback)callback done:(B
   }
 }
 
-- (BOOL)readAllParts:(RCTMultipartCallback)callback
+- (void)emitProgress:(NSDictionary *)headers
+       contentLength:(NSUInteger)contentLength
+               final:(BOOL)final
+            callback:(RCTMultipartProgressCallback)callback
+{
+  if (headers == nil) {
+    return;
+  }
+  // Throttle progress events so we don't send more that around 60 per second.
+  CFTimeInterval currentTime = CACurrentMediaTime();
+
+  NSUInteger headersContentLength = headers[@"Content-Length"] != nil ? [headers[@"Content-Length"] unsignedIntValue] : 0;
+  if (callback && (currentTime - _lastDownloadProgress > 0.016 || final)) {
+    _lastDownloadProgress = currentTime;
+    callback(headers, @(headersContentLength), @(contentLength));
+  }
+}
+
+- (BOOL)readAllParts:(RCTMultipartCallback)callback progressCallback:(RCTMultipartProgressCallback)progressCallback
```

Comment:
Oups forgot to update the signature in tests, also updated it to `readAllPartsWithCompletionCallback:progressCallback:`. Should land this time!